src/driver/cockroachdb/CockroachQueryRunner.ts

```diff
@@ -1336,7 +1336,7 @@ export class CockroachQueryRunner extends BaseQueryRunner implements QueryRunner
         }).join(" OR ");
 
         const constraintsSql = `SELECT "ns"."nspname" AS "table_schema", "t"."relname" AS "table_name", "cnst"."conname" AS "constraint_name", ` +
-            `CASE "cnst"."contype" WHEN 'x' THEN pg_get_constraintdef("cnst"."oid", true) ELSE "cnst"."consrc" END AS "expression", ` +
+            `CASE "cnst"."contype" WHEN 'c' THEN regexp_replace(pg_get_constraintdef("cnst"."oid", true), '^CHECK\\s*\\((.*)\\)\\s*$', '\\1', 'i') ELSE pg_get_constraintdef("cnst"."oid", true) END AS "expression", ` +
```

Comment:
Why do we have to use regexp here? Can't we just update the place where this value is read?

Reply:
I was trying to write a minimally invasive patch, but it's probably nicer to do the replacement later when building the `TableCheck`. Will provide a patch for that.

Reply:
Can this be resolved? Replacing the regular expression by some other parsing is probably not worth it and storing the constriaint definition including `CHECK(...)` would require changes in many other places which handle constraints.